Makefile

```diff
@@ -84,5 +88,17 @@ njs-unit-test: ## Run unit tests for the njs httpmatches module
 		node:18 \
 		/bin/bash -c "npm install && npm test && npm run clean"
 
+.PHONY: generate-njs-yaml
+generate-njs-yaml:
+	kubectl create configmap njs-modules --from-file=$(NJS_DIR)/httpmatches.js --dry-run=client --output=yaml > $(MANIFEST_DIR)/njs-modules.yaml
+
+.PHONY: fetch-crds-yaml
+fetch-crds-yaml:
```

Comment:
A user would also need to manually upgrade these, but may not want to upgrade both our CRDs and gateway CRDs if they both live in our `crds/` directory.

Reply:
I tested this out - helm will ignore anything that's already present on the cluster on both install and upgrade. So, if the user wanted to install NKG with our CRDs, and they already had the Gateway API resources installed on the cluster, helm will install any not-present CRDs and leave the other resources unmodified. The user may then want to upgrade the Gateway API resources, which they can do from the source in our Chart, or directly from the Gateway repo. I've updated the documentation to add some information around when to update the Gateway APIs manually, and how to do it. Additionally, if they want to upgrade these resources separately, we can add documentation on how to do so when we have our own CRDs. They'll share a directory, but should have their own files. Ideally though they'll want to upgrade both on an upgrade, as we have a dependency on a particular version of the Gateway APIs.

Reply:
I also removed the `--skip-crds` modifier from the commands, as we require the Gateway CRDs to be present on the cluster (and we know now they will be untouched by Helm if they are already present). If in the future we want to provide the option to skip installing our own CRDs, we can provide instructions to do so with the caveat that the Gateway API resources will need to be installed separately.

Reply:
The crds folder also includes Gateway API RBAC, webhook and few more things, so it is not really CRDs. I just wonder if we misusing the feature of helm.

Reply:
My aim was to make it as easy as possible for users to get up and running, but we can exclude the Gateway resources and have the extra manual step if we have concerns about the approach?

Reply:
I might lean towards the extra manual step of installing Gateway first (and therefore also manually upgrading). While slightly less convenient, it leaves Gateway resources out of our hands and responsibilities. It is a bit hacky to include them in the `crds/` folder.

Reply:
From learnings with NIC customers, I recommend sorting out how to include them magically.
And also how to allow Helm to upgrade them.
Since the CRDs are required, and not shipping with K8s core, not including the matching version is kind of a downer for the user.

Reply:
Helm can't upgrade anything that's in the `crds/` folder. It's meant to be used for CRDs that we define and ship for our product specifically (like when we define them for configuring our control plane or nginx).

The only way we can get Helm to upgrade Gateway resources is if we include them directly in our chart templates, which is against best practices for Helm due to unpredictable and unsupported behavior. We also become the source of truth for Gateway resources, instead of the Gateway repo itself, which is not ideal. Increases the risk of going out of sync. Also creates a dependency on us for a user to upgrade their Gateway resources if a patch is pushed, because we would have to update and re-release our helm charts, versus a user being able to just upgrade their Gateway resources directly.

Reply:
Ideally, at some point Gateway will release their own helm chart, and we can create a dependency on that.

Reply:
I agree with @sjberman. I think the risks of including the Gateway API CRDs far outweigh the benefits. Asking the user to run one additional command to install the Gateway API CRDs does not seem like a big ask.